pkg/oauth/oauth.go

```diff
@@ -0,0 +1,452 @@
+		ClientID:     config.ClientID,
+		ClientSecret: config.ClientSecret,
+		AccessToken:  tokens.AccessToken,
+		RefreshToken: tokens.RefreshToken,
+		TokenType:    tokens.TokenType,
+		Expiry:       tokens.Expiry,
+	}
+	primarySecretStorage := utils.KeyRingStorage{User: login}
+	fallbackSecretStorage := utils.FileSystemStorage{BaseDir: jiraDir}
+
+	if err := utils.SaveJSON(primarySecretStorage, oauthSecretsFile, oauthSecrets); err != nil {
+		if errors.Is(err, keyring.ErrSetDataTooBig) {
+			cmdutil.Warn("Data was too big to save to the keyring, falling back to filesystem storage")
+		}
+		err = utils.SaveJSON(fallbackSecretStorage, oauthSecretsFile, oauthSecrets)
+		if err != nil {
+			return nil, fmt.Errorf("failed to store OAuth secrets: %w", err)
+		}
+		cmdutil.Warn("Saved credentials to owner-restricted filesystem storage")
+	}
+
+	return &ConfigureTokenResponse{
+		AccessToken:  tokens.AccessToken,
+		RefreshToken: tokens.RefreshToken,
+		CloudID:      cloudID,
+	}, nil
+}
+
+// LoadOAuthSecrets loads OAuth secrets from storage.
+func LoadOAuthSecrets(login string) (*OAuthSecrets, error) {
+	jiraDir, err := getJiraConfigDir()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get Jira config directory: %w", err)
+	}
+	primaryStorage := utils.KeyRingStorage{User: login}
+	fallbackSecretStorage := utils.FileSystemStorage{BaseDir: jiraDir}
+	secrets, err := utils.LoadJSON[OAuthSecrets](primaryStorage, oauthSecretsFile)
+	if err != nil {
+		fmt.Printf("Warning: Primary storage failed to save, using fallback")
+		secrets, err = utils.LoadJSON[OAuthSecrets](fallbackSecretStorage, oauthSecretsFile)
+		if err != nil {
+			return nil, fmt.Errorf("failed to load OAuth secrets: %w", err)
+		}
+	}
+
+	return &secrets, nil
+}
+
+// HasOAuthCredentials checks if OAuth credentials are present.
+func HasOAuthCredentials(login string) bool {
+	_, err := LoadOAuthSecrets(login)
+	return err == nil
+}
+
+// collectOAuthCredentials collects OAuth credentials from the user.
+func collectOAuthCredentials() (*OAuthConfig, error) {
+	answers := struct {
+		ClientID     string
+		ClientSecret string
+		RedirectURI  string
+	}{}
+
+	// Check for environment variables
+	envClientID := os.Getenv("JIRA_CLI_OAUTH_CLIENT_ID")
+	envClientSecret := os.Getenv("JIRA_CLI_OAUTH_CLIENT_SECRET")
+
+	q1 := &survey.Input{
+		Message: "Jira App Client ID:",
+		Help:    "This is the client ID of your Jira App that you created for OAuth authentication.",
+		Default: envClientID,
+	}
+
+	q2 := &survey.Password{
+		Message: "Jira App Client Secret:",
+		Help:    "This is the client secret of your Jira App that you created for OAuth authentication.",
+	}
+
+	q3 := &survey.Input{
+		Default: defaultRedirectURI,
+		Message: "Redirect URI:",
+		Help:    "The redirect URL for Jira App. Recommended to set as localhost.",
+	}
+
+	if envClientID == "" {
+		if err := survey.AskOne(q1, &answers.ClientID); err != nil {
+			return nil, err
+		}
+	} else {
+		answers.ClientID = envClientID
+	}
```

Comment:
I would have written this to be simpler


```suggestion
	answers.ClientID = envClientID
	if envClientID == "" {
		if err := survey.AskOne(q1, &answers.ClientID); err != nil {
			return nil, err
		}
	}
```

Same for other blocks